api/data_explorer/controllers/export_url_controller.py

```diff
@@ -336,28 +244,89 @@ def _append_to_query(existing, new, join, table_num):
+            # underscores here and periods in table_name attribute.
+            'name': table_name.replace('.', '_').replace(':', '_'),
+            'attributes': {
+                'dataset_name': current_app.config['DATASET_NAME'],
+                'table_name': table_name,
+            }
+        })
 
-    query = _get_sql_query(filter_arr)
-    cohort_name = data['cohortName']
     for c in ' .:=':
         cohort_name = cohort_name.replace(c, '_')
+    entities.append({
+        'entityType': 'cohort',
+        'name': cohort_name,
+        'attributes': {
+            'dataset_name': current_app.config['DATASET_NAME'],
+            'query': _get_sql_query(filter_arr),
+            'data_explorer_url': data_explorer_url,
+        }
+    })
+
+    return entities
+
+
+def _write_gcs_file(entities):
+    """Returns GCS file path of the format /bucket/object."""
+    client = storage.Client(project=current_app.config['DEPLOY_PROJECT_ID'])
+    export_bucket = client.get_bucket(
+        current_app.config['EXPORT_URL_GCS_BUCKET'])
+    user = os.environ.get('USER')
+    random_str = ''.join(
+        random.choice(string.ascii_letters + string.digits) for _ in range(10))
+    blob = export_bucket.blob(random_str)
+    entities_json = json.dumps(entities)
+    blob.upload_from_string(entities_json)
+
+    current_app.logger.info(
+        'Wrote gs://%s/%s' %
+        (current_app.config['EXPORT_URL_GCS_BUCKET'], blob.name))
+    # Return in the format that signing a URL needs.
+    return '/%s/%s' % (current_app.config['EXPORT_URL_GCS_BUCKET'], blob.name)
+
+
+def _create_signed_url(gcs_path):
+    private_key_path = os.path.join(current_app.config['DATASET_CONFIG_DIR'],
+                                    'private-key.json')
+    creds = ServiceAccountCredentials.from_json_keyfile_name(private_key_path)
+    service_account_email = current_app.config[
+        'DEPLOY_PROJECT_ID'] + '@appspot.gserviceaccount.com'
+    # Signed URL will be valid for 5 minutes
+    timestamp = str(int(time.time()) + 5 * 60)
+    file_metadata = '\n'.join(['GET', '', '', timestamp, gcs_path])
+    signature = base64.b64encode(creds.sign_blob(file_metadata)[1])
+    signature = urllib.quote(signature, safe='')
+    signed_url = ('https://storage.googleapis.com%s?GoogleAccessId=%s'
+                  '&Expires=%s&Signature=%s') % (
+                      gcs_path, service_account_email, timestamp, signature)
+    # import-data expects url to be url encoded
+    signed_url = urllib.quote(signed_url, safe='')
+    current_app.logger.info('Signed URL: ' + signed_url)
+    return signed_url
 
-    entities = _get_entities_dict(cohort_name, query, filter_arr,
-                                  data_explorer_url)
 
-    # Don't actually write GCS file during unit test. If we wrote a file during
-    # unit test, in order to make it easy for anyone to run this test, we would
-    # have to create a world-readable bucket.
-    if 'pytest' in sys.modules:
```

Comment:
Why is this being removed? Did we remove the relevant unit tests? I forget.

Reply:
Yes #262 

Only test left is https://github.com/DataBiosphere/data-explorer/blob/master/api/data_explorer/test/test_elasticsearch_util.py